### PR TITLE
Allow blood type to take any values

### PR DESCRIPTION
**Pull request by sibinhho99-nus**

### CODE REVIEW

File: src/main/java/seedu/address/model/medical/BloodType.java [hunk 1]
```diff
@@ -7,17 +7,17 @@
...
 
-    public static final String MESSAGE_CONSTRAINTS = "Blood type must be either A, B, AB or O.";
+    public static final String MESSAGE_CONSTRAINTS = "Blood type can take any values, and it should not be blank.";
 
     /*
-     * Blood type must be either A, B, AB, O, or nil (in case user did not specify it).
+     * Blood type can take any value
      */
-    public static final String VALIDATION_REGEX = "[A|B|O|AB|nil]";
+    public static final String VALIDATION_REGEX = "[^\\s].*";
```
Comment (clement0010) [inline src/main/java/seedu/address/model/medical/BloodType.java:20]: Why is this the case? I don't think it's appropriate to not validate blood type input.

Comment (sibinhho99-nus) [reply src/main/java/seedu/address/model/medical/BloodType.java:20]: There are currently 43 possible blood group systems which is hard to account for ([https://en.wikipedia.org/wiki/Human_blood_group_systems](https://en.wikipedia.org/wiki/Human_blood_group_systems))

Comment (clement0010) [reply src/main/java/seedu/address/model/medical/BloodType.java:20]: Why not just narrow it down into **one** system. I don't think other group systems are commonly used in hospitals (project scope)

Comment (sibinhho99-nus) [reply src/main/java/seedu/address/model/medical/BloodType.java:20]: > Why not just narrow it down into **one** system -- ABO. I don't think other group systems are commonly used in hospitals (project scope)
> 
> image

What if a patient with a rare blood type visits the hospital? We cannot reject his or her treatment or force an incorrect blood type onto him or her that might cause incompatibility when doing transfusion. It's better to give the user the flexibility to write down any extra information they might need

Comment (clement0010) [reply src/main/java/seedu/address/model/medical/BloodType.java:20]: Can u give an example other than ABO? rhesus factor can be validated as well.

Comment (clement0010) [reply src/main/java/seedu/address/model/medical/BloodType.java:20]: Anyways I think this is a small thing lets not spend too much time discussing this.

**Review by clement0010 — APPROVED** [review]

**Pull request merged**